backend/inspirehep/disambiguation/tasks.py

```diff
@@ -70,26 +75,127 @@ def disambiguate_signatures(self, clusters):
     db.session.commit()
 
 
+def match_literature_author_with_config(author_data, matcher_config):
+    matched_records = [
+        matched_record
+        for matched_record in match(author_data, matcher_config)
+        if matched_record.get("inner_hits")
+    ]
+    matched_authors_refs = match_literature_author_ids(matched_records)
+    return matched_authors_refs, matched_records
+
+
+def match_literature_author_with_validator(validators, author_data, matched_references):
+    for validator in validators:
+        valid_matches = (
+            match for match in matched_references if validator(author_data, match)
+        )
+        matched_valid_authors_references = match_literature_author_ids(valid_matches)
+        if matched_valid_authors_references:
+            return matched_valid_authors_references
+
+
+def match_literature_author_ids(matched_records):
```

Comment:
naming is confusing, as this doesn't do any match, it postprocesses matches and returns the matching refs, maybe something like
```suggestion
def get_record_ref_if_unambiguous_match(matched_records):
```